### PR TITLE
SRRS Integration Tests

### PR DESCRIPTION
**Pull request by ieperen3039**
tests the SRRS as a black-box

### CODE REVIEW

File: cmd/srrs/main_test.go [hunk 1]
```diff
@@ -109,25 +122,86 @@ func TestAll(t *testing.T) {
...
+	// command SRRC -> TRC
+	for i := 0; i < 10; i++ {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			c := apitest.RandomCommand()
+			state := &api.State{
+				Command: *c,
+			}
+
+			log.Debug("Sending command %s SRRC -> TRC", c)
+			err = wsConn.WriteJSON(state)
```
Comment (rvolosatovs) [inline cmd/srrs/main_test.go]: That's not how it works. WebSocket is read-only. States/commands are sent via REST API, see the spec on drive.

**Comment by ieperen3039** [comment]
I am not sure whether the http requests are used correctly, but I have to do my USE project this weekend as well. If this must be merged before the planning session, please feel free to make the changes and merge it yourself

File: cmd/srrs/main_test.go [hunk 2]
```diff
@@ -109,25 +124,90 @@ func TestAll(t *testing.T) {
...
+
+	// command SRRC -> TRC
+	for i := 0; i < 10; i++ {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			c := apitest.RandomCommand()
+
+			log.Debug("Sending command %s SRRC -> TRC", c)
+			pld, err := json.Marshal(c)
+			a.NoError(err)
+			req, err := http.NewRequest(http.MethodPut, "https://"+defaultAddr+"/api/v1/command", bytes.NewReader(pld))
```
Comment (rvolosatovs) [inline cmd/srrs/main_test.go]: Don't hardcode the endpoint, use variable from `main`

File: cmd/srrs/main_test.go [hunk 3]
```diff
@@ -109,25 +124,90 @@ func TestAll(t *testing.T) {
...
+			a.Nil(err)
+			a.Equal(got.Command, c)
+			log.Debug("Random state received on WebSocket")
+		})
+	}
+
+	// command SRRC -> TRC
+	for i := 0; i < 10; i++ {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			c := apitest.RandomCommand()
```
Comment (rvolosatovs) [inline cmd/srrs/main_test.go]: maybe `cmd` is a bit clearer?

File: cmd/srrs/main_test.go [hunk 4]
```diff
@@ -36,19 +39,27 @@ func init() {
...
+			logger.Fatalf("Failed to open Unix socket on %s: %s", unixSockPath, err)
 		}
-	}
 
-	netLst, err = net.Listen("unix", unixSockPath)
-	if err != nil {
-		logger.Fatalf("Failed to open Unix socket on %s: %s", unixSockPath, err)
+	} else {
+		logger.Debug("Creating TCP socket")
+		netLst, err = net.Listen("tcp", *tcpSock)
```
Comment (golangci) [inline cmd/srrs/main_test.go]: ineffectual assignment to `err`

**Comment by rvolosatovs** [comment]
Added in #102

**Pull request closed without merging**